app/views/AttachmentView.js

```diff
@@ -146,22 +147,57 @@ class AttachmentView extends React.Component {
 		);
 	}
 
-	renderVideo = uri => (
-		<Video
-			source={{ uri }}
-			rate={1.0}
-			volume={1.0}
-			isMuted={false}
-			resizeMode={Video.RESIZE_MODE_CONTAIN}
-			shouldPlay
-			isLooping={false}
-			style={styles.container}
-			useNativeControls
-			onLoad={() => this.setState({ loading: false })}
-			onError={console.log}
-			ref={this.getVideoRef}
-		/>
-	);
+	checkDocumentDir=async() => {
+		const { attachment } = this.state;
+		const {
+			image_url, image_type, video_url, video_type
+		} = attachment;
+		const url = image_url || video_url;
+
+		if (isAndroid) {
+			const rationale = {
+				title: I18n.t('Write_External_Permission'),
+				message: I18n.t('Write_External_Permission_Message')
+			};
+			const result = await PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE, rationale);
+			if (!(result || result === PermissionsAndroid.RESULTS.GRANTED)) {
+				return;
+			}
+		}
+
+		try {
+			const extension = image_url ? `.${ mime.extension(image_type) || 'jpg' }` : `.${ mime.extension(video_type) || 'mp4' }`;
+			const fileName = `${ SHA256(url) + extension }`;
+			const files = await FileSystem.readDirectoryAsync(FileSystem.documentDirectory);
+			files.forEach((t) => {
+				if (t === fileName) {
+					this.setState({ path: `${ FileSystem.documentDirectory + fileName }` });
+				}
+			});
+		} catch (e) {
+			EventEmitter.emit(LISTENER, { message: I18n.t(image_url ? 'error-save-image' : 'error-save-video') });
+		}
+	}
```

Comment:
This function is pretty much like `handleSave`, can you refactor it so we can extract what we need for both?